### DISCUSSION TITLE
[JIT] Orientatiom

### DISCUSSION THREAD

**Discussion by ianwijma** [tailwindlabs/tailwindcss#4764] [Ideas]
Now with the new JIT compiler it would be cool to have a media query for orientation. Which utilises the [orientation @media](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/orientation). 

## Reason

When working on implementing a design, creating something that works on a iPhone 5/CE (320x568) is a challenge. Especially when the user views the content in landscape mode. Then you are working with 320 pixel is height. Which is some cases is quite tricky.

## Solution 1 - adding a new utility

While looking at this idea, I was looking at existing issues where something has been suggested a year ago #1397, but because of css size constraints, this was not viable. Now we have the JIT compilation. We might want to revisit this idea.

Here is a code example, that tackles the issue in the section above. Where I had a major space constraints on a small device. Which resizes content depending on the devices size.

```jsx
<div className='landscape:h-10 landscape:sm:h-16'>
  ... Unnecessary content here
</div>
```

## Solution 2 - adding a custom media query

As always there are different solutions to a problem. Where in this case as mentioned in #1397, you can just add the custom media query yourself. But I think now with the JIT compiler. it would be cool to have all utilities available out of the box. requiring less and less configuration for the end user.

## Final words

Let me know if there is ANY interest for this as I would LOVE to contribute any code to this project! LOVE the idea and execution behind TailwindCSS and I would love to give back!

**Comment by lukewarlow** [comment]
https://github.com/tailwindlabs/tailwindcss/pull/6046 - made a PR to add an `orientation-portrait:` and `orientation-landscape:` variant to Tailwind.
